packages/network/p2p/manager.go

```diff
@@ -172,21 +172,11 @@ func (m *Manager) Send(packet proto.Message, protocolID string, to ...identity.I
 	}
 
 	for _, nbr := range neighbors {
-		stream := nbr.GetStream(protocol.ID(protocolID))
-		if stream == nil {
-			m.log.Warnw("send error, no stream for protocol", "peer-id", nbr.ID(), "protocol", protocolID)
-			nbr.Close()
-			continue
-		}
-		if err := stream.WritePacket(packet); err != nil {
-			m.log.Warnw("send error", "peer-id", nbr.ID(), "err", err)
-			nbr.Close()
-		}
-
+		nbr.Enqueue(packet, protocol.ID(protocolID))
 		receivers = append(receivers, nbr.ID())
 	}
 
-	return
+	return receivers
```

Comment:
Why are we copying receivers here again? First of all the return value is never used and `m.AllNeighbors()` and `m.GetNeighborsByID(to)` already return a copy

Reply:
Removed it.